meteor/server/migration/1_42_0.ts

```diff
@@ -9,6 +9,15 @@ export const addSteps = addMigrationSteps('1.42.0', [
 		id: 'Add new routeType property to routeSets where missing',
 		canBeRunAutomatically: true,
 		validate: async () => {
+			// If routeSets has been converted to ObjectWithOverrides,
+			// it will have a defaults property, and shouln't be migrated
+			if (
+				(await Studios.countDocuments({
+					routeSetsWithOverrides: { $exists: true },
+				})) > 0
+			) {
+				return false
+			}
```

Comment:
I think you are correct in this, but I think that even this is flawed, as is the original implementation.

For reference, this migration has been unchanged since being added in r42.
It is checking for any studio is missing `routeSets`. It would then do a `Object.entries(studio.routeSets)` for each studio, which in order for the migration to be run means that a studio is missing `studio.routeSets`. Which means that the call to `Object.entries(studio.routeSets)` will throw for that studio.

So if this migration added in r42 was ever run, it would throw an error and fail. It is worth noting that there was a migration added in r24 that would ensure that the routeSets property existed.

So I'm going to remove this migration instead of fixing it, as it appears to have never worked as intended, and hasn't been missed.

Reply:
Yes, that sounds reasonable as well 👍